gcloud-java-core/src/main/java/com/google/gcloud/ServiceOptions.java

```diff
@@ -111,12 +115,22 @@ public HttpTransport create() {
       }
       // Consider Compute
       try {
-        return AuthCredentials.getComputeCredential().getTransport();
+        return getComputeHttpTransport();
```

Comment:
I have the feeling that this is not necessary (but this needs to be validated).
Basically, I think `NetHttpTransport` works on GCE (consider [this](https://github.com/google/google-auth-library-java/blob/281d485e679545ed06f46e6090fc5d47f4a140e8/oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java#L45) and [this](https://github.com/google/google-api-java-client/blob/3d866eb7094cfd31ad9f7acefbadd6cd5171be48/google-api-client/src/main/java/com/google/api/client/googleapis/javanet/GoogleNetHttpTransport.java#L55)).

And the the when choosing the default auth (if not specified) it will be checked (see [this](https://github.com/google/google-auth-library-java/blob/281d485e679545ed06f46e6090fc5d47f4a140e8/oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java#L98).

If we need a way to check "isOnGCE" we can apply as similar check to the link above (but I am not sure we need one).
 
Please validate it with both default and service account authentication.